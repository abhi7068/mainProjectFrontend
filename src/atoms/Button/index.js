import React from 'react';
import classes from './button.module.css';

export default function Button() {
  return (
    <div>
      <button data-testid='button' className={classes.buttonRed}>This is a normal Button</button>
    </div>
  );
}
