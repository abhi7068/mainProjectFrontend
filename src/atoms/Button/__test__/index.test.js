import React from 'react';

import Button from '../index';
import { render } from '@testing-library/react';

it('it renders', () => {
  render(<Button />);
});
it('should take a snapshot', () => {
  const { asFragment } = render(<Button />);
  expect(asFragment(render(<Button />))).toMatchSnapshot();
});
it('it renders with correct text', () => {
  const { getByTestId } = render(<Button />);
  expect(getByTestId('button')).toHaveTextContent('This is a normal Button');
});
