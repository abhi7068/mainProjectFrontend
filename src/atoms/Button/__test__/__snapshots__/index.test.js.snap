// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should take a snapshot 1`] = `
<DocumentFragment>
  <div>
    <button
      class="buttonRed"
      data-testid="button"
    >
      This is a normal Button
    </button>
  </div>
</DocumentFragment>
`;
